[18.0] [MIG] stock_account_move_reset_to_draft: Migration to 18.0

ivantodorovich opened the pull request:
It includes:
- https://github.com/OCA/account-invoicing/pull/2064

victoralmau committed:
[ADD] stock_account_move_multi_reset_to_draft: New module

TT51201

mymage committed:
Added translation using Weblate (Italian)

kanda999 committed:
[FIX] stock_account_move_reset_to_draft: Update the README
Corrected inaccurate information in the README.

leemannd committed:
[IMP] stock_account_move_reset_to_draft: pre-commit auto fixes

leemannd committed:
[MIG] stock_account_move_reset_to_draft: Migration to 17.0

AungKoKoLin1997 committed:
[FIX] stock_account_move_reset_to_draft: only consider generated SVLs from account move line

ivantodorovich committed:
[MIG] stock_account_move_reset_to_draft: Migration to 18.0

ivantodorovich committed:
[IMP] stock_account_move_reset_to_draft: assert the exception message

ivantodorovich committed:
[IMP] stock_account_move_reset_to_draft: drop use of Form for faster test execution

`Form` isn't actually needed here

grindtildeath commented on stock_account_move_reset_to_draft/models/account_move.py:
```diff
@@ -0,0 +1,75 @@
+        the sum of the value in opposite sign.
+        """
+        for item in self.sudo().filtered(
+            lambda x: x.is_inbound
```
```suggestion
            lambda x: x.is_inbound()
```
`is_inbound` being a function it would always evaluate to True if not called.

ivantodorovich replied:
Indeed, good catch!

Moreover `is_inbound` was not the right method as that's based on the money flow.
The correct one is `is_purchase_document`, so I changed it

grindtildeath commented on stock_account_move_reset_to_draft/models/account_move.py:
```diff
@@ -0,0 +1,75 @@
+        """
+        for item in self.sudo().filtered(
+            lambda x: x.is_inbound
+            and any(line.stock_valuation_layer_ids for line in x.line_ids)
+        ):
+            for line in item.line_ids.filtered("stock_valuation_layer_ids"):
```
I feel like the condition in first if is unneeded with the filtering from the for loop

ivantodorovich committed:
[FIX] stock_account_move_reset_to_draft: filtering conditions

fix: properly call is_purchase_document() in place of is_inbound, which was not
called and not correct.

imp: redundant filtering on stock_valuation_layer_ids

AungKoKoLin1997 reviewed (approved):
Code Review: LGTM

grindtildeath commented:
/ocabot merge nobump
